cairo-contracts/packages/clients/src/cometbft/component.cairo

```diff
@@ -415,11 +415,42 @@ pub mod CometClientComponent {
 
             let header_height = header.clone().signed_header.height;
 
-            // TODO: Implement consensus state pruning mechanism.
+            let update_heights = self.read_update_heights(client_sequence);
+            let mut update_heights_span = update_heights.span();
 
-            if !self.consensus_state_exists(client_sequence, header_height.clone()) {
-                let mut client_state = self.read_client_state(client_sequence);
+            let mut client_state = self.read_client_state(client_sequence);
+
+            let mut heights_kept = array![];
+            let mut check_in_progress = true;
+            // Since th Heights are sorted when stored, as soon as we find the first
+            // Height which isn't expired we can stop checking the rest and build
+            // the new Heights array which are kept.
+            while let Option::Some(height) = update_heights_span.pop_front() {
+                if check_in_progress {
+                    let consensus_state = self
+                        .read_consensus_state(client_sequence, height.clone());
+
+                    let expiry_timestamp = consensus_state.timestamp.as_secs()
+                        + client_state.trusting_period.seconds;
+
+                    let host_timestamp = get_block_timestamp();
+
+                    if expiry_timestamp <= host_timestamp {
```

Comment:
you can use the prexisting `consensus_state.state(client_state.trusting_period, Duration::zero())` to check if the consensus state is expired.

Reply:
Done in cb64954b61ade3a79def6d6a03f31bc760cdadb6